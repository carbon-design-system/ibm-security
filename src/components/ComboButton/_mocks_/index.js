/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
