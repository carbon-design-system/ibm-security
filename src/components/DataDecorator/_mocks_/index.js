/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

const props = {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
const midLine = {
  type: 'MD5',
  value: '0f3deda483df5e5f8043ea20297d243b',
  score: 30,
  href: '#',
};

export default props;

export { props, midLine };
