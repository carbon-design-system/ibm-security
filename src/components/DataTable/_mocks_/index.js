/* eslint-disable react/prop-types */
/**
 * @file Data table mocks.
 * @copyright IBM Security 2019
 */

import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
} from '../../..';

const headers = [
  {
    key: 'lastSighted',
    header: 'Last sighted',
  },
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'type',
    header: 'Type',
  },
  {
    key: 'attackPhase',
    header: 'Attack phase',
  },
  {
    key: 'method',
    header: 'Method',
  },
  {
    key: 'missing',
    header: 'Missing',
  },
];

const isSelectable = false;
const isSortable = false;

const rows = [
  {
    id: '7c34f4507156b35',
    lastSighted: '11/28/2018',
    name: 'IBM X-Force',
    type: 'IPv6',
    attackPhase: 'Initial Compromise',
    method: 'Phishing',
  },
  {
    id: '38347816a51b0c5a8',
    lastSighted: '06/28/2017',
    name: 'AppX',
    type: 'Hash',
    attackPhase: 'Establish Foothold',
    method: 'Deploy backdoor',
  },
  {
    id: '1d6116a2d1d03b4',
    lastSighted: '05/30/2018',
    name: 'Bigfix',
    type: 'URL',
    attackPhase: 'Escalate Privileges',
    method: 'Credential dumping',
  },
  {
    id: 'c52b0419981221765',
    lastSighted: '12/12/2018',
    name: 'Am I Affected',
    type: 'Hash',
    attackPhase: 'Cross-Site Scripting',
    method: 'Malicious Code',
  },
  {
    id: '697626e6f703131d',
    lastSighted: '11/11/2018',
    name: 'QRadar',
    type: 'Hash',
    attackPhase: 'SQL Injection Attack',
    method: 'SQL Vulnerabilities',
  },
  {
    id: 'd568f12fa7b7cac1',
    lastSighted: '12/09/2018',
    name: 'iris',
    type: 'Hash',
    attackPhase: 'Volume Based Attacks',
    method: 'ICMP floods',
  },
  {
    id: '697626e6f70313554',
    lastSighted: '12/10/2018',
    name: 'TISP',
    type: 'Hash',
    attackPhase: 'Application Layer Attacks',
    method: 'GET/POST floods',
  },
  {
    id: '1dd568f12fa7b7cac2',
    lastSighted: '12/11/2018',
    name: 'CLX',
    type: 'Hash',
    attackPhase: 'Protocol Attacks',
    method: 'SYN Floods',
  },
  {
    id: '697626e6f703131',
    lastSighted: '10/12/2018',
    name: 'X-Force Plus',
    type: 'Hash',
    attackPhase: 'Malicious Links',
    method: 'Social Engineering',
  },
  {
    id: 'dd568f12fa7b7cac0',
    lastSighted: '12/16/2018',
    name: 'BigFix & iris',
    type: 'Hash',
    attackPhase: 'Exploit Kits',
    method: 'Scripts',
  },
  {
    id: '697626e6f70313',
    lastSighted: '01/12/2018',
    name: 'Cases',
    type: 'Hash',
    attackPhase: 'Malicious Email',
    method: 'Social Engineering',
  },
];

const page = 1;
const pageSize = 5;
const pageSizes = [5, 10, 20];

/**
 * Render method.
 * @returns {string} The string to render.
 */
const render = ({
  rows,
  headers,
  getHeaderProps,
  getRowProps,
  getTableProps,
  getSelectionProps,
}) => (
  <TableContainer>
    <Table {...getTableProps()}>
      {headers && (
        <TableHead>
          <TableRow>
            <TableSelectAll {...getSelectionProps()} />
            {headers.map(header => (
              <TableHeader
                key={header.id}
                {...getHeaderProps({ header, isSortable })}
              >
                {header.header}
              </TableHeader>
            ))}
          </TableRow>
        </TableHead>
      )}
      <TableBody>
        {rows.map(row => (
          <TableRow {...getRowProps({ row })} key={row.id}>
            <TableSelectRow {...getSelectionProps({ row })} />
            {row.cells.map(cell => (
              <TableCell key={cell.id}>
                {cell.value ? cell.value : 'â€“'}
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </TableContainer>
);

export {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  render,
  rows,
};
