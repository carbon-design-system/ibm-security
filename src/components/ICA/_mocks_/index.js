/**
 * @file Important content area mocks.
 * @copyright IBM Security 2019
 */

const value = 10;

export default {
  label: 'Label',
  value,
  total: value * value,
};
