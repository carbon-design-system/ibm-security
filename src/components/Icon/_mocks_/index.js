/**
 * @file Icon mocks.
 * @copyright IBM Security 2019 - 2021
 */

import { Add16 } from '@carbon/icons-react';

const path = 'M0';
const renderIcon = Add16;

export { path, renderIcon };
