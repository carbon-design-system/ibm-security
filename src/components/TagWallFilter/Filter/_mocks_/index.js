/**
 * @file Filter mocks.
 * @copyright IBM Security 2018
 */

const items = [
  {
    id: 'item-1',
    label: 'New York',
  },
  {
    id: 'item-2',
    label: 'London',
  },
  {
    id: 'item-3',
    label: 'Tokyo',
  },
  {
    id: 'item-4',
    label: 'Beijing',
  },
  {
    id: 'item-5',
    label: 'New Delhi',
  },
  {
    id: 'item-6',
    label: 'Berlin',
  },
  {
    id: 'item-7',
    label: 'Montreal',
  },
  {
    id: 'item-8',
    label: 'San Fransisco',
  },
  {
    id: 'item-9',
    label: 'Rio de Janeiro',
  },
  {
    id: 'item-10',
    label: 'Johannesburg',
  },
  {
    id: 'item-11',
    label: 'Abuja',
  },
];

const itemToString = (item) => (item ? item.label : '');

export { items, itemToString };
