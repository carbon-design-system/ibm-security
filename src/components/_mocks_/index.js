/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
