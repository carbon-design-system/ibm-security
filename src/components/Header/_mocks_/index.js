/**
 * @file Header mocks.
 * @copyright IBM Security 2018
 */

import { url } from '../../_mocks_';
import labels from '../locales/en/Header.json';

const hasSearch = true;
const hasSearchError = false;

const links = {
  product: url(),
  profile: url(),
  registration: url(),
  sign_in: url(),
  sign_out: url(),
  edit_profile: url(),
  notifications_preferences: url(),
  notifications_view_all: url(),
};

const accounts = new Array(3).fill().map((name = `Company`, id) => ({
  id: id.toString(),
  name: `${name} ${id + 1}`,
}));

const notifications = [
  {
    id: `notification__0`,
    description: 'A new request requires approval.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Dec 21 2018 18:57:08 GMT-0600 (Central Standard Time)',
    label: '11min ago',
  },
  {
    id: `notification__1`,
    description: 'There have been 17 failed sign-ins in the last 3 days.',
    href: url(),
    product: 'Application 2',
    datetime: 'Wed May 01 2019 22:22:52 GMT-0500 (Central Daylight Time)',
    label: '1hr ago',
  },
  {
    id: `notification__2`,
    description: 'There are 3 unread messages in the queue.',
    href: url(),
    product: 'Application 3',
    datetime: 'Fri Jul 06 2018 00:17:49 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__3`,
    description: 'Review the results of the latest scan.',
    href: url(),
    product: 'Application 3',
    datetime: 'Sun Apr 14 2019 20:32:17 GMT-0500 (Central Daylight Time)',
    label: '3hrs ago',
  },
  {
    id: `notification__4`,
    description: 'A new administrator has been added to your account.',
    href: url(),
    product: 'Application 1',
    datetime: 'Sat Oct 20 2018 19:07:04 GMT-0500 (Central Daylight Time)',
    label: '4hrs ago',
  },
  {
    id: `notification__5`,
    description: 'There are 5 items that require your review.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Dec 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '5hrs ago',
  },
  {
    id: `notification__7`,
    description:
      'There are 103 threats found in the last scan. Please click to see the details.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Nov 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '10hrs ago',
  },
  {
    id: `notification__8`,
    description:
      'There are 3 ongoing attacks. Please click to take actions immediately.',
    href: url(),
    product: 'Application 2',
    datetime: 'Tue Oct 25 2018 18:14:28 GMT-0600 (Central Standard Time)',
    label: '24hrs ago',
  },
];

const profile = {
  image_url: null,
  name: {
    first_name: 'Sample',
    surname: 'User',
  },
};

const profileWithAccount = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: accounts[0],
});

const profileWithAccountLongName = Object.assign({}, profile, {
  email: 'example@ibm.com',
  account: {
    ...accounts[0],
    name: `${accounts[0].name} plus more text to make the name super super long`,
  },
});
const defaultSearchValue = '127.0.0.1';

export {
  defaultSearchValue,
  hasSearch,
  hasSearchError,
  labels,
  links,
  accounts,
  notifications,
  profile,
  profileWithAccount,
  profileWithAccountLongName,
};
