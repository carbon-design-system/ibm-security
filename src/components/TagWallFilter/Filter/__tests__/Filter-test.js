/**
 * @file Filter tests.
 * @copyright IBM Security 2018
 */

import { shallow } from 'enzyme';
import React from 'react';

import { label, className } from '../../../_mocks_';
import { items, itemToString } from '../_mocks_';

import FilterRaw from '../Filter';

const { fn } = jest;

describe('Filter', () => {
  let filter;
  let onChange;

  beforeEach(() => {
    onChange = fn();

    filter = shallow(
      <FilterRaw
        className={className}
        id="test-id"
        items={items}
        itemToString={itemToString}
        onChange={onChange}
        placeholder={label}
      />
    );
  });

  describe('render', () => {
    it('renders correctly', () => {
      expect(filter).toMatchSnapshot();
    });

    it("renders the HTML of the node's subtree", () => {
      expect(filter.render()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should invoke `onChange` when the filter is changed', () => {
      filter.simulate('change');
      expect(onChange).toBeCalled();
    });
  });
});
