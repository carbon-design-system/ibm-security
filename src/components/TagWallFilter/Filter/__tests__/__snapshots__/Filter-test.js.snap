// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter render renders correctly 1`] = `
<Downshift
  breakingChanges={Object {}}
  clearSelection={[Function]}
  defaultHighlightedIndex={null}
  defaultInputValue=""
  defaultIsOpen={false}
  defaultSelectedItem={null}
  environment={[Window]}
  getA11yStatusMessage={[Function]}
  highlightedIndex={null}
  inputValue=""
  isOpen={true}
  itemToString={[Function]}
  onChange={[Function]}
  onInputValueChange={[Function]}
  onOuterClick={[Function]}
  onSelect={[Function]}
  onStateChange={[Function]}
  onUserAction={[Function]}
  render={[Function]}
  selectedItemChanged={[Function]}
  stateReducer={[Function]}
/>
`;

exports[`Filter render renders the HTML of the node's subtree 1`] = `
<div
  class="security--filter bx--list-box"
  id="test-id"
>
  <div
    aria-expanded="true"
    aria-haspopup="true"
    aria-label="close menu"
    class="bx--list-box__field"
    data-toggle="true"
    id="test-id"
    role="button"
    tabindex="-1"
    type="button"
  >
    <input
      aria-autocomplete="list"
      aria-expanded="true"
      autocomplete="off"
      class="security--filter__input-field bx--text-input"
      id="test-id"
      placeholder="Label"
      role="combobox"
      value=""
    />
    <span
      class="security--filter__search"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--filter__search__icon"
        fill="var(--cds-icon-01, #f4f4f4)"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        style="will-change:transform"
        viewBox="0 0 32 32"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M29,27.5859l-7.5521-7.5521a11.0177,11.0177,0,1,0-1.4141,1.4141L27.5859,29ZM4,13a9,9,0,1,1,9,9A9.01,9.01,0,0,1,4,13Z"
        />
      </svg>
    </span>
  </div>
  <div
    class="security--filter__list-container"
    id="test-id"
    role="listbox"
  >
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-0"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-0"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Abuja"
          >
            Abuja
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-1"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-1"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Beijing"
          >
            Beijing
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-2"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-2"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Berlin"
          >
            Berlin
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-3"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-3"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Johannesburg"
          >
            Johannesburg
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-4"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-4"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="London"
          >
            London
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-5"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-5"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Montreal"
          >
            Montreal
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-6"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-6"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="New Delhi"
          >
            New Delhi
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-7"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-7"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="New York"
          >
            New York
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-8"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-8"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Rio de Janeiro"
          >
            Rio de Janeiro
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-9"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-9"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="San Fransisco"
          >
            San Fransisco
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--list-box__menu-item"
      id="downshift-0-item-10"
      role="button"
      tabindex="0"
    >
      <div
        class="bx--list-box__menu-item__option"
      >
        <div
          aria-labelledby="downshift-0-item-10"
          class="security--filter__list-item__entry"
        >
          <span
            class="bx--text-truncate--end"
            title="Tokyo"
          >
            Tokyo
          </span>
          <span
            class="security--filter__add"
          >
            <svg
              aria-hidden="true"
              class="security--icon security--filter__add__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              style="will-change:transform"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
